### PR TITLE
test:e2e: create Security Alert Blockaid tests for eth_sendTransaction requests and no PPOM response request

### PR DESCRIPTION
**Pull request by digiwand**
## **Description**

Adds e2e tests for PPOM Blockaid Security Alert for various benign and malicious eth_sendTransaction requests. We also test for the case the PPOM request fails and we display a warning to the user. 

**Adds additional functions:** 
- FixtureBuilder#withNetworkControllerOnMainnet
- test/e2e/mock-server-json-rpc.ts - helper method to mock JSON-RPC requests

**Note:** 
Some cases are not provided in this PR 
e.g. SetApprovalForAll and ERC721 transfer. 
See https://github.com/MetaMask/MetaMask-planning/issues/991 for more details. 

**Author comment:** 
I'm actually not the biggest fan of the test names. In this PR we have: 
- snaps/ppom-blockaid-alert
- snaps/ppom-malicious-erc20-approval
- snaps/ppom-malicious-erc20-transfer

I think when we move these files outside of the snaps folder, it would be nice to update the methods to e.g. 
- snaps/ppom-blockaid-alert
- snaps/ppom-blockaid-alert-malicious-erc20-approval
- snaps/ppom-blockaid-alert-malicious-erc20-transfer

The reason for my thought is that PPOM stands for Privacy Preserving Offline Module which IMO is generic and doesn't encompass all of our Blockaid logic. I view the logic as Blockaid being the umbrella with PPOM, UI, and requests following underneath it.

## **Manual testing steps**

Run:
- yarn test:e2e:single test/e2e/snaps/ppom-blockaid-alert.spec.js --browser=chrome
- yarn test:e2e:single test/e2e/snaps/ppom-malicious-erc20-approval.spec.js --browser=chrome
- yarn test:e2e:single test/e2e/snaps/ppom-malicious-erc20-transfer.spec.js --browser=chrome

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/991
Blocked By: https://github.com/MetaMask/metamask-extension/pull/21034

### CODE REVIEW

File: test/mocks/json-rpc-result.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
+const MOCK_BLOCK_NUMBER = '0x1';
```
Comment (jpuri) [inline test/mocks/json-rpc-result.ts:1]: Hey, just curious, will these response require change if data returned by CDN changes.

**Review by jpuri — DISMISSED** [review]
🚀

File: test/e2e/snaps/ppom-malicious-erc20-transfer.spec.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  openDapp,
+  unlockWallet,
+  withFixtures,
+} = require('../helpers');
+
+const bannerAlertSelector = '[data-testid="security-provider-banner-alert"]';
+
+const selectedAddress = '0x5cfe73b6021e818b776b421b1c4db2474086a7e1';
+const selectedAddressWithoutPrefix = '5cfe73b6021e818b776b421b1c4db2474086a7e1';
+const USDC_ADDRESS = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48';
```
Comment (seaona) [inline test/e2e/snaps/ppom-malicious-erc20-transfer.spec.js:16]: nit: we could move this inside the CONTRACT_ADDRESS object?

**Review by seaona — DISMISSED** [review]

File: test/mocks/json-rpc-result.ts (hunk 1 above)
Comment (digiwand) [reply test/mocks/json-rpc-result.ts:1]: not exactly sure what data you're referring to. Do you have an example? I have observed changing this value may affect tests in the `snaps/ppom-blockaid-alert`

File: test/e2e/snaps/ppom-malicious-erc20-transfer.spec.js (hunk 2 above)
Comment (digiwand) [reply test/e2e/snaps/ppom-malicious-erc20-transfer.spec.js]: updated [d0f19d4](https://github.com/MetaMask/metamask-extension/pull/21086/commits/d0f19d4361132baa2478f67d76409dacea1a1246)

File: test/e2e/mock-server-json-rpc.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ *    ['eth_gasPrice],
+ *    ['eth_getBalance', {
+ *      result: '0x27d17a5b79f77509541',
+ *    }],
+ *    ['eth_getBlockByNumber']
+ *  ]);
+ * ```
+ */
+async function mockServerJsonRpc(
+  mockServer: any, // MockttpServer
```
Comment (Mrtenz) [inline test/e2e/mock-server-json-rpc.ts:42]: Can this be typed?

Comment (digiwand) [reply test/e2e/mock-server-json-rpc.ts]: thx for the nudge @Mrtenz!

updated [46d77c5](https://github.com/MetaMask/metamask-extension/pull/21086/commits/46d77c58dbeafb43d71c2d8a956d7b13328b516e)

**Review by Mrtenz — APPROVED** [review]

**Review by seaona — APPROVED** [review]

**Comment by digiwand** [comment]
- create tests/e2e/snaps/ppom-malicious-erc20-approval.spec.js
- create tests/e2e/snaps/ppom-malicious-erc20-transfer.spec.js
- create helper method FixtureBuilder#withNetworkControllerOnMainnet
- create test/e2e/mock-server-json-rpc.ts - helper function to mock JSON-RPC requests

**Pull request merged**